script/readyhelper.js

```diff
@@ -0,0 +1,83 @@
+define(
+  'bigscreenplayer/readyhelper', [
+    'bigscreenplayer/models/mediastate',
+    'bigscreenplayer/models/windowtypes',
+    'bigscreenplayer/models/livesupport'
+  ],
+  function (MediaState, WindowTypes, LiveSupport) {
+    var ReadyHelper = function (initialPlaybackTime, windowType, liveSupport, callback) {
+      var ready = false;
+
+      var callbackWhenReady = function (evt) {
+        if (ready) return;
+
+        if (!evt.data) {
+          ready = false;
+        } else if (evt.timeUpdate) {
+          ready = isValidTime(evt.data);
+        } else {
+          ready = isValidState(evt.data) && isValidTime(evt.data);
+        }
+
+        if (ready) {
+          callback();
+        }
+      };
+
+      function isValidState (evtData) {
+        return evtData.state && evtData.state !== MediaState.FATAL_ERROR;
+      }
+
+      function isValidTime (evtData) {
+        var isStatic = windowType === WindowTypes.STATIC;
+
+        if (isStatic) {
+          return validateStaticTime(evtData.currentTime);
+        } else {
+          return validateLiveTime(evtData.currentTime, evtData.seekableRange);
+        }
+      }
+
+      function validateStaticTime (currentTime) {
+        if (currentTime !== undefined) {
+          if (initialPlaybackTime) {
+            return currentTime > 0;
+          } else {
+            return currentTime >= 0;
+          }
+        }
+        return false;
+      }
+
+      function validateLiveTime (currentTime, seekableRange) {
+        if (liveSupport === LiveSupport.PLAYABLE) {
+          return currentTime >= 0;
+        }
+
+        if (isValidSeekableRange(seekableRange)) {
+          var currTimeGtStart = currentTime >= seekableRange.start;
```

Comment:
currentTimeWithinRange feels a more suitable option since we are not interested in the variables individually.

Reply:
do you mean pulling the range checking logic into a separate function?

Reply:
No, just a single variable instead of two.

Reply:
We split it up because it was really long as one variable, and went quite a bit off the side of the screen

Reply:
OK, I condensed it down, I think it's still readable enough; I think at one point the logic was a bit more complex there